packages/neuron-ui/src/components/AmendSend/hooks.ts

```diff
@@ -0,0 +1,159 @@
+import React, { useState, useCallback, useEffect, useMemo } from 'react'
+import { TFunction } from 'i18next'
+import { AppActions, StateDispatch } from 'states/stateProvider/reducer'
+import { getTransaction as getOnChainTransaction } from 'services/chain'
+import { getTransaction as getSentTransaction, getTransactionSize } from 'services/remote'
+import { isSuccessResponse } from 'utils'
+
+const clear = (dispatch: StateDispatch) => {
+  dispatch({
+    type: AppActions.ClearSendState,
+  })
+}
+
+const useUpdateTransactionPrice = (dispatch: StateDispatch) =>
+  useCallback(
+    (value: string) => {
+      dispatch({
+        type: AppActions.UpdateSendPrice,
+        payload: value,
+      })
+    },
+    [dispatch]
+  )
+
+const useSendDescriptionChange = (dispatch: StateDispatch) =>
+  useCallback(
+    (e: React.SyntheticEvent<HTMLInputElement>) => {
+      const { value } = e.target as HTMLInputElement
+      dispatch({
+        type: AppActions.UpdateSendDescription,
+        payload: value,
+      })
+    },
+    [dispatch]
+  )
+
+export const useInitialize = ({
+  hash,
+  walletID,
+  price,
+  dispatch,
+}: {
+  hash: string
+  walletID: string
+  price: string
+  isMainnet: boolean
+  dispatch: React.Dispatch<any>
```

Comment:
https://github.com/nervosnetwork/neuron/blob/6dc256395412fe64b967bd5704675bfecdff01cd/packages/neuron-ui/src/states/stateProvider/reducer.ts#L125

maybe we can replace the dispatch with the above one

Reply:
This PR is ready to merge except this unresolved conversation, any idea for this @devchenyan

Reply:
> This PR is ready to merge except this unresolved conversation, any idea for this @devchenyan

I've got an omission. Solved